### PR TITLE
fix issue #512 to save the file location as a table column

### PR DESCRIPTION
**Pull request by jackyq2015**

### CODE REVIEW

**Comment by magdmartin** [comment]
fix for #512

File: main/src/com/google/refine/importers/tree/ImportParameters.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+            boolean includeFileSources, String fileSource) {
+        this.trimStrings = trimStrings;
+        this.storeEmptyStrings = storeEmptyStrings;
+        this.guessDataType = guessCellValueTypes;
+        this.includeFileSources = includeFileSources;
+        this.fileSource = fileSource;
+    }
+
+    public ImportParameters(boolean trimStrings, boolean storeEmptyStrings, boolean guessCellValueTypes) {
+        this.trimStrings = trimStrings;
```
Comment (tfmorris) [inline main/src/com/google/refine/importers/tree/ImportParameters.java]: This should call the constructor above (Line 11) rather than duplicating code.

File: main/src/com/google/refine/importers/tree/ImportParameters.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        this.storeEmptyStrings = storeEmptyStrings;
+        this.guessDataType = guessCellValueTypes;
+        this.includeFileSources = false;
+        this.fileSource = "";
+    }
+    
+}
```
Comment (tfmorris) [inline main/src/com/google/refine/importers/tree/ImportParameters.java]: missing new line

File: main/src/com/google/refine/importers/tree/XmlImportUtilities.java [hunk 3]
```diff
@@ -282,18 +269,6 @@ static public void importTreeData(
         }
     }
 
-    @Deprecated
```
Comment (tfmorris) [inline main/src/com/google/refine/importers/tree/XmlImportUtilities.java]: Removing deprecated methods shouldn't be included in this patch.

File: main/src/com/google/refine/importers/tree/TreeImportingParserBase.java [hunk 4]
```diff
@@ -210,8 +214,22 @@ protected void parseOneFile(
         boolean trimStrings = JSONUtilities.getBoolean(options, "trimStrings", true);
         boolean storeEmptyStrings = JSONUtilities.getBoolean(options, "storeEmptyStrings", false);
         boolean guessCellValueTypes = JSONUtilities.getBoolean(options, "guessCellValueTypes", true);
-
-        XmlImportUtilities.importTreeData(treeParser, project, recordPath, rootColumnGroup, limit2, trimStrings,
-                storeEmptyStrings,guessCellValueTypes);
+        
+        // copied from TabularImportingParserBase
```
Comment (tfmorris) [inline main/src/com/google/refine/importers/tree/TreeImportingParserBase.java]: If this is used in both Tabular & Tree imports, is there a reason it can't be hoisted to ImportingParserBase?

File: main/src/com/google/refine/importers/tree/ImportParameters.java (hunk 1 above)
Comment (jackyq2015) [inline main/src/com/google/refine/importers/tree/ImportParameters.java]: Agree, do you want me to commit the change or create another branch from the fork?

File: main/src/com/google/refine/importers/tree/TreeImportingParserBase.java (hunk 4 above)
Comment (jackyq2015) [inline main/src/com/google/refine/importers/tree/TreeImportingParserBase.java]: "copied from somewhere" always can be improved. will revisit the room to pull up to parent class.

**Comment by tfmorris** [comment]
I've incorporated fixes for most of these comments in be936a86eb77f664341f0dd70b2174f9d1fe8b7c.  I'll discuss with @magdmartin who he had review the changes before he agreed to commit the PR to see if we can improve the process.
